cornell-dti/course-plan
# Walkthrough update

### Summary 

Update the walkthrough steps to highlight new CoursePlan features, update the styling to match the finalized designs, and fix some walkthrough bugs. (See [notion](https://www.notion.so/courseplan/82bf355c4b7b49f99691eb9f3afb203c?v=5da9d646f3ef424bbceb1ca0e8e00cf8))

![image](https://user-images.githubusercontent.com/25535093/111073878-ea045980-84b6-11eb-988e-cbc23b6f9669.png)

![image](https://user-images.githubusercontent.com/25535093/111019251-fc35a900-838b-11eb-88e5-020a5fdbfe14.png)

- [x] Updated steps to focus on requirements menu
- [x] Wrote specific case to open requirements menu and PE requirement for second step
- [x] Updated TourWindow designs with new text/assets for the first and final modals
- [x] Updated tooltip designs (button styling, progress count, etc.)
- [x] Fixed bug where walkthrough would pop up every time you clicked finish on the last page of the onboarding modal
- [x] Fixed styling bug where the semester view would show up under the walkthrough if large enough, see below image

![image](https://user-images.githubusercontent.com/25535093/111019132-38b4d500-838b-11eb-9dc8-29694316f8bf.png)



Remaining TODOs:

- [x] Update styling again to match designs updated after previous commits (mainly `TourWindow` changes and adding a 4th step)
- [ ] Make step 1 and 2 of the tooltips more seamless/differentiated. I suggested to @einc only having the PE course section highlighted in the second step, but was unable to get a tooltip to show up on that subrequirement div, as tooltips only show up if they are in the html when the tour is started. This tooltip would not be bound to that html element until the reqs menu was opened fully, and thus did not show up in the tour even after opened. Could explore more.
- [x] Was getting the follow error in my build until I set the tour object to type `any` instead of the default `IntroJS`: `Property '_currentStep' does not exist on type 'IntroJs'.` Is there a better way of handling this? I also had to put a comment to stop eslint from breaking on the use of an underscore property, but that seems to be the way to access it from IntroJS.

### Test Plan 

1. Confirm that the onboarding modal now cannot be opened by finishing onboarding a second time, but deleting your firestore data and going through onboarding will have it show up.

To test the rest of the changes, change the if statement in the `endOnboarding()` function of `Dashboard.vue` to `if (true)` locally so that you can open the walkthrough by clicking through the Onboarding modal.

2. Confirm that with semesters extending off the bottom of the screen, the walkthrough modal now displays correctly
3. Click through walkthrough steps and make sure it works as expected and looks like designs
4. Skip the walkthrough during the PE step and make sure you can collapse and reopen that requirement
